src/lib.rs

```diff
@@ -244,6 +193,57 @@ fn run_rustpython(vm: &VirtualMachine, run_mode: RunMode) -> PyResult<()> {
             }
         }
     }
+    #[cfg(feature = "flame-it")]
+    {
+        main_guard.end();
+        if let Err(e) = write_profile(&vm.state.as_ref().settings) {
+            error!("Error writing profile information: {}", e);
+        }
+    }
+    Ok(())
+}
+
+#[cfg(feature = "flame-it")]
+fn write_profile(settings: &Settings) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Moved this further down, right above the tests so it doesn't obstruct the rest of the code.